Scripts: Use webbrowser package in Python to open a URL in a web browser

LumioseSil opened the pull request:
#### 82bd766bd89d5a7865183ae72be484db7e27126f
<pre>
Scripts: Use webbrowser package in Python to open a URL in a web browser
<a href="https://bugs.webkit.org/show_bug.cgi?id=253515">https://bugs.webkit.org/show_bug.cgi?id=253515</a>

Reviewed by Jonathan Bedard.

I made a mistake. Because &quot;start&quot; worked on the command line, I assumed
that it would work under the Terminal.run() command. Now I realize this
is not what actually happens. Instead of reverting that commit, commit
<a href="https://commits.webkit.org/261332@main">https://commits.webkit.org/261332@main</a>, I am instead introducing a
proper fix, which is using the URL and opening it in a browser via the
webbrowser package.

* Tools\Scripts\libraries\webkitcorepy\webkitcorepy\terminal.py: Change
  &apos;start&apos; back to &apos;explorer.&apos; Teach open_url to call the webbrowser
  url method to open webpages if the scheme is http or https.

* Tools\Scripts\webkitpy\common\system\user.py: Inline redundant method.

Canonical link: <a href="https://commits.webkit.org/261687@main">https://commits.webkit.org/261687@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/375cb7b415df3297fb99029a7af7decb802515e9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112571 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21722 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/1229 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4347 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/121121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/116637 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/23062 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/12884 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/86/builds/5498 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118338 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/17132 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100341 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105637 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/115992 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/99078 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/885 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/46141 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/14058 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/921 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/95240 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14741 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/10295 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/111334 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/20081 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52926 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8150 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16577 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:
```diff
@@ -544,7 +545,9 @@ def main(cls, args, repository, **kwargs):
                 info_url = 'https://github.com/WebKit/WebKit/wiki/Git-Config#Configuration-Options'
                 print('For detailed information about the options configured by this script, please see:\n{}'.format(info_url))
                 if not args.defaults and Terminal.choose("Would you like to open this URL in your browser?", default='Yes') == 'Yes':
-                    if not Terminal.open_url(info_url):
```
The point of `Terminal.open_url` is that code like this can be insulated from decisions we make about how URLs should be opened. If we were going to make this change, it would be in `terminal.py`, not here here.

AZero13 replied:
This brings us back to the original problem of Windows opening an explorer tab, not a web browser

JonWBedard reviewed (commented):
I need to do a bit of digging about how this package works, and if we want to include a dependency. We have a system for this, but it's a bit different than what most projects use (our goal is to not modify the global Python environment, but also, have scripts "just work" when you run them)

JonWBedard reviewed (changes requested):
I'm marking this as "Request changes", not because I expect @AtariDreams to change anything at the moment, but because I'm going to be out of town for a few days and importing a new package like this looks innocent enough, but until we vet this package and add it to our auto installer, this PR could break things that are not visible from EWS. Other operations folks familiar with our Python imports like @aj062 and @ryanhaddad could vet this also, but are likely to defer to me. I don't want someone to just drive-by review this in the next couple days.

I will absolutely be giving better guidance on a path forward (and perhaps just an r+, depending on what the vetting reveals) when I get back on the 14th.

JonWBedard reviewed (approved):
Looks like `webbrowser` is part of Python standard libraries, which means we don't have to bother with ensuring the package is installed.

On an unrelated note, @AtariDreams, it looks like you are periodically rebasing this change (and presumably other changes as well). That's only necessary if there is a merge conflict that cannot be automatically resolved, our infrastructure knows how to rebase changes before landing. This is one of the main features of [`merge-queue`](https://github.com/WebKit/WebKit/wiki/Contributing#landing-changes).

LumioseSil committed:
Scripts: Use webbrowser package in Python to open a URL in a web browser
https://bugs.webkit.org/show_bug.cgi?id=253515

Reviewed by Jonathan Bedard.

I made a mistake. Because "start" worked on the command line, I assumed
that it would work under the Terminal.run() command. Now I realize this
is not what actually happens. Instead of reverting that commit, commit
https://commits.webkit.org/261332@main, I am instead introducing a
proper fix, which is using the URL and opening it in a browser via the
webbrowser package.

* Tools\Scripts\libraries\webkitcorepy\webkitcorepy\terminal.py: Change
  'start' back to 'explorer.' Teach open_url to call the webbrowser
  url method to open webpages if the scheme is http or https.

* Tools\Scripts\webkitpy\common\system\user.py: Inline redundant method.

Canonical link: https://commits.webkit.org/261687@main
